odiglet/pkg/kube/runtime_details/workload_enabled_predicated.go

```diff
@@ -0,0 +1,105 @@
+package runtime_details
+
+import (
+	"github.com/odigos-io/odigos/common/consts"
+	appsv1 "k8s.io/api/apps/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+// this predicate is used for workload reconciler, and will only pass events
+// where the workload is changed to odigos instrumentation enabled.
+// This way, we don't need to run language detection downstream when unnecessary.
+// This also helps in managing race conditions, where we might re-add runtime details
+// which were just deleted by instrumentor controller and generate unnecessary noise
+// in the k8s eventual consistency model.
+type WorkloadEnabledPredicate struct {
+	predicate.Funcs
+}
+
+func (i *WorkloadEnabledPredicate) Create(e event.CreateEvent) bool {
+	enabled := isInstrumentationEnabled(e.Object)
+	// only handle new workloads if they start with instrumentation enabled
+	return enabled
+}
+
+func (i *WorkloadEnabledPredicate) Update(e event.UpdateEvent) bool {
+
+	if e.ObjectOld == nil {
+		return false
+	}
+	if e.ObjectNew == nil {
+		return false
+	}
+
+	// only run runtime inspection if the workload was not instrumented before
+	// and now it is.
+	oldEnabled := isInstrumentationEnabled(e.ObjectOld)
+	newEnabled := isInstrumentationEnabled(e.ObjectNew)
+	becameEnabled := !oldEnabled && newEnabled
```

Comment:
Maybe we can add here
`if becameEnabled {
       return true
}`
and avoid checking it in the following special cases for deployments, statfulsets and daemonsets